apps/arweave/src/ar_http_iface_server.erl

```diff
@@ -64,128 +76,95 @@ start() ->
 		Semaphores
 	),
 	ok = ar_blacklist_middleware:start(),
-	ok = start_http_iface_listener(Config),
-	ok = start_gateway_listeners(Config),
+	ok = case ar_config:should_enable_https() of
+		true ->
+			start_https_iface_listener(Config);
+		_ ->
+			ok
+	end,
+	ok = case ar_config:should_disable_http() of
+		false ->
+			start_http_iface_listener(Config);
+		_ ->
+			ok
+	end,
 	ok.
 
 start_http_iface_listener(Config) ->
 	Dispatch = cowboy_router:compile([{'_', ?HTTP_IFACE_ROUTES}]),
-	HttpIfaceEnv = #{ dispatch => Dispatch },
 	TransportOpts = #{
 		max_connections => Config#config.max_connections,
 		socket_opts => [
-                  {port, Config#config.port},
-                  {keepalive, true}
-                ]
+			{port, Config#config.port},
+			{keepalive, true}
+		]
 	},
-	ProtocolOpts = protocol_opts([{http_iface, HttpIfaceEnv}]),
-	{ok, _} = cowboy:start_clear(ar_http_iface_listener, TransportOpts, ProtocolOpts),
-	ok.
-
-stop() ->
-	lists:foreach(fun(Listener) ->
-		try cowboy:stop_listener(Listener) of
-			_ -> ok
-		catch
-			_:_ -> ok
-		end
-	end, [ar_http_iface_listener, ar_http_gateway_listener, ar_https_gateway_listener]).
-
-start_gateway_listeners(Config) ->
-	case Config#config.gateway_domain of
-		Domain when is_binary(Domain) ->
-			ok = start_http_gateway_listener(Config),
-			ok = start_https_gateway_listener(Config),
-			ok;
-		not_set ->
-			ok
-	end.
-
-start_http_gateway_listener(Config) ->
-	Domain = Config#config.gateway_domain,
-	CustomDomains = Config#config.gateway_custom_domains,
-	TransportOpts = #{
-		max_connections => 100,
-		num_acceptors => 100,
-		socket_opts => [{port, 80}]
+	ProtocolOpts = #{
+		inactivity_timeout => 120000,
+		idle_timeout => 30000,
+		middlewares => ?HTTP_IFACE_MIDDLEWARES,
+		env => #{ dispatch => Dispatch },
+		metrics_callback => fun prometheus_cowboy2_instrumenter:observe/1,
+		stream_handlers => [cowboy_metrics_h, cowboy_stream_h]
 	},
-	ProtocolOpts = protocol_opts([{gateway, Domain, CustomDomains}]),
-	{ok, _} = cowboy:start_clear(ar_http_gateway_listener, TransportOpts, ProtocolOpts),
+	{ok, _} = cowboy:start_clear(ar_http_iface_listener, TransportOpts, ProtocolOpts),
 	ok.
 
-start_https_gateway_listener(Config) ->
-	PrivDir = code:priv_dir(arweave),
-	Domain = Config#config.gateway_domain,
-	CustomDomains = Config#config.gateway_custom_domains,
-	Dispatch = cowboy_router:compile([{'_', ?HTTP_IFACE_ROUTES}]),
-	HttpIfaceEnv = #{ dispatch => Dispatch },
-	SniHosts = derive_sni_hosts(CustomDomains),
+start_https_iface_listener(Config) ->
+	DataDir = Config#config.data_dir,
+	AlternativeHostnames = Config#config.alternative_hostnames,
+	CertfilePath = filename:join([DataDir, "tls/cert.pem"]),
+	KeyfilePath = filename:join([DataDir, "tls/key.pem"]),
```

Comment:
+TODO add watcher which will monitor changes of certificate files, because let's encrypt certbot can silently modify those files so your HTTPS will be outdated in several months because renewed and stored cert is not used by your app until restart